authors/apps/authentication/serializers.py

```diff
@@ -147,3 +152,56 @@ def update(self, instance, validated_data):
         instance.save()
 
         return instance
+
+class SocialAuthSerializer(serializers.Serializer):
+    auth_token = serializers.CharField()
+    
+    def validate_auth_token(self, auth_token):
+        user_data = self.get_user_data(auth_token)
+        return user_data
+
+
+class FacebookAuthSerializer(SocialAuthSerializer):
+    
+    def get_user_data(self, auth_token):
+        """
+        This function decodes the received token into data required from the user, 
+        all this with the help of facebook's GraphAPI. Data received includes email,
+        name and profile picture
+        """
+        try:
+            graph = facebook.GraphAPI(access_token=auth_token)
+            user_profile = graph.request('/me?fields=name,email,picture')
+            return user_profile
+        except:
+            return 'Please provide a valid token'
+                
+
+class GoogleAuthSerializer(SocialAuthSerializer):
+
+    def get_user_data(self, auth_token):
+        try:
+            user_profile = id_token.verify_oauth2_token(auth_token, requests.Request())
+            if user_profile['iss'] not in ['accounts.google.com', 'https://accounts.google.com']:
+                raise ValueError('Wrong issuer.')
+            return user_profile            
+        except:
+            return 'Please provide a valid token'
+
+
+class TwitterAuthSerializer(SocialAuthSerializer):
+
+    def get_user_data(self, auth_token):
+        consumer_key = "N5lUnIborTryBvGqnA9KKKl5N"
+        consumer_secret = "MCxkmUDaIJkcj80tKGK7NC55BMXtX0uP6GrUnhfuK2TQaYe6Hj"
```

Comment:
when you get the chance, you decouple here

Reply:
Thanks @dannylwe, Have worked on it